src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -714,22 +719,67 @@ class ResourceService(
     accessPolicyDAO.loadPolicy(policyId, samRequestContext)
 
-  def loadResourcePolicy(policyIdentity: FullyQualifiedPolicyId, samRequestContext: SamRequestContext): IO[Option[AccessPolicyMembership]] =
+  def loadResourcePolicy(policyIdentity: FullyQualifiedPolicyId, samRequestContext: SamRequestContext): IO[Option[AccessPolicyMembershipResponse]] =
     accessPolicyDAO.loadPolicyMembership(policyIdentity, samRequestContext)
 
   private def makeValidatablePolicies(
-      policies: Map[AccessPolicyName, AccessPolicyMembership],
+      policies: Map[AccessPolicyName, AccessPolicyMembershipRequest],
       samRequestContext: SamRequestContext
   ): IO[Set[ValidatableAccessPolicy]] =
     policies.toList
-      .traverse { case (accessPolicyName, accessPolicyMembership) =>
-        makeCreatablePolicy(accessPolicyName, accessPolicyMembership, samRequestContext)
+      .traverse { case (accessPolicyName, accessPolicyMembershipRequest) =>
+        val maybeOwnerPolicyEmails = getPolicyEmailsFromAccessPolicyMembership(accessPolicyMembershipRequest, samRequestContext)
+        val allEmails = collateMemberAndPolicyEmails(accessPolicyMembershipRequest.memberEmails, maybeOwnerPolicyEmails)
+        for {
+          newMemberEmails <- allEmails
+          validatablePolicy <- makeValidatablePolicy(
+            accessPolicyName,
+            accessPolicyMembershipRequest.copy(memberEmails = newMemberEmails),
+            samRequestContext
+          )
+        } yield validatablePolicy
       }
       .map(_.toSet)
 
-  private def makeCreatablePolicy(
+  private def getPolicyEmailsFromAccessPolicyMembership(
+      accessPolicyMembership: AccessPolicyMembershipRequest,
+      samRequestContext: SamRequestContext
+  ): Option[IO[Set[WorkbenchEmail]]] = {
+    val maybePolicyEmails = accessPolicyMembership.memberPolicies.map { memberPolicy =>
+      memberPolicy.map { memberPolicy =>
+        val ownerPolicy = loadPolicyByPolicyIdentifiers(memberPolicy, samRequestContext)
+        ownerPolicy.map(p => p.map(_.email))
+      }
+    }
+    // this mainly serves to simplify the complex structure of policy emails
+    maybePolicyEmails.map { policyEmails =>
+      val listIOPolicyEmails = policyEmails.toList
+      val ioListPolicyEmails = listIOPolicyEmails.sequence
+      val ioSetPolicyEmails = for {
+        listPolicyEmails <- ioListPolicyEmails
+      } yield listPolicyEmails.flatten.toSet
+      ioSetPolicyEmails
+    }
+  }
+
+  private def loadPolicyByPolicyIdentifiers(policyIdentifiers: PolicyIdentifiers, samRequestContext: SamRequestContext): IO[Option[AccessPolicy]] =
+    accessPolicyDAO.loadPolicy(
+      FullyQualifiedPolicyId(FullyQualifiedResourceId(policyIdentifiers.resourceTypeName, policyIdentifiers.resourceId), policyIdentifiers.policyName),
+      samRequestContext
+    )
+
+  private def collateMemberAndPolicyEmails(memberEmails: Set[WorkbenchEmail], maybePolicyEmails: Option[IO[Set[WorkbenchEmail]]]): IO[Set[WorkbenchEmail]] =
```

Comment:
I don't think this needs to be its own function. You could just tack on the contents of this function onto the end of `getPolicyEmailsFromAccessPolicyMembership` and get the same result. That way, we're maintaining `IO` as the top-level type, and this function isn't used anywhere else.